src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,45 @@
+package core.basesyntax;
+
+import core.basesyntax.figure.*;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private final Random random = new Random();
+    private final ColorSupplier supplier = new ColorSupplier();
+    public Figure getRandomFigure() {
+        Color color = supplier.getRandomColor();
+        return switch (random.nextInt(5)) {
```

Comment:
Avoid using magic numbers directly in your code. Instead, define a constant for the number of figure types. For example, `public static final int FIGURE_COUNT = 5;` and use it in `random.nextInt(FIGURE_COUNT)`.